core/app/models/concerns/spree/user_api_authentication.rb

```diff
@@ -2,6 +2,21 @@
 
 module Spree
   module UserApiAuthentication
+    # Generate a json web token
+    # @see https://github.com/jwt/ruby-jwt
+    # @return [String]
+    def generate_jwt_token(options: Spree::Config.jwt_options, expires_in: nil)
+      # @see https://github.com/jwt/ruby-jwt#support-for-reserved-claim-names
+      extras = {}
+      extras['exp'] = Time.current.to_i + expires_in if expires_in.present?
+      extras['iat'] = Time.current
+
+      payload = as_json(options)
+      payload.merge(extras)
+
+      JWT.encode payload, Spree::Config.jwt_secret, Spree::Config.jwt_algorithm
+    end
```

Comment:
who saves this token on the User?

Reply:
The token never gets saved to the database and has no need to.  It is encrypted with a secret, (By default Rails.application.secret_key_base). If the token fails to decrypt it is unauthentic or has been tampered with.